opensrp-app/src/main/java/org/smartregister/util/JsonFormUtils.java

```diff
@@ -809,20 +809,22 @@ public static void fillSubFormAddressFields(JSONObject jsonObject, Map<String, A
 
     // Helper functions
 
+    /**
+     * Refactored for backward compatibility invokes getMultiStepFormFields which provides the same result
+     * Returns a JSONArray of all the forms fields in a single or multi step form.
+     *
+     * @param jsonForm {@link JSONObject}
+     * @return fields {@link JSONArray}
+     */
     public static JSONArray fields(JSONObject jsonForm) {
         try {
 
-            JSONObject step1 = jsonForm.has(STEP1) ? jsonForm.getJSONObject(STEP1) : null;
-            if (step1 == null) {
-                return null;
-            }
-
-            return step1.has(FIELDS) ? step1.getJSONArray(FIELDS) : null;
+            return getMultiStepFormFields(jsonForm);
```

Comment:
This method has added complexity than the method being removed if the form has only 1 step. Can `getMultiStepFormFields` be refactored so that if a form has 1 step then ` step1.getJSONArray(FIELDS) ` is used

Reply:
Figured that to be a negligible trade off given 1. its duplicating the exact same functionality and 2. in the case that it is only one step in the form being processed, the loop only executes once.